x-pack/plugin/fleet/src/internalClusterTest/java/org/elasticsearch/xpack/fleet/action/GetGlobalCheckpointsActionIT.java

```diff
@@ -305,42 +304,63 @@ public void testWaitOnIndexCreated() throws Exception {
         assertFalse(response.timedOut());
     }
 
-    public void testPrimaryShardsNotReadyNoWait() throws Exception {
-        final GetGlobalCheckpointsAction.Request request = new GetGlobalCheckpointsAction.Request(
-            "not-assigned",
-            false,
-            false,
-            EMPTY_ARRAY,
-            TEN_SECONDS
-        );
+    /**
+     * Cluster remains yellow when initial primary is THROTTLED (and unavailable) during creation.
+     * This test verifies that implementation can handle this scenario.
+     */
+    public void testWaitOnIndexCreatedWithThrottling() {
+
+        client().admin()
+            .cluster()
+            .prepareUpdateSettings()
+            .setPersistentSettings(
+                Settings.builder().put(CLUSTER_ROUTING_ALLOCATION_NODE_INITIAL_PRIMARIES_RECOVERIES_SETTING.getKey(), 0).build()
+            )
+            .get();
+
         client().admin()
             .indices()
-            .prepareCreate("not-assigned")
+            .prepareCreate("throttled-during-creation")
             .setWaitForActiveShards(ActiveShardCount.NONE)
             .setSettings(
                 Settings.builder()
                     .put(IndexSettings.INDEX_TRANSLOG_DURABILITY_SETTING.getKey(), Translog.Durability.REQUEST)
                     .put("index.number_of_shards", 1)
                     .put("index.number_of_replicas", 0)
-                    .put(IndexMetadata.INDEX_ROUTING_INCLUDE_GROUP_SETTING.getKey() + "node", "none")
             )
             .get();
 
-        UnavailableShardsException exception = expectThrows(
-            UnavailableShardsException.class,
-            () -> client().execute(GetGlobalCheckpointsAction.INSTANCE, request).actionGet()
-        );
-        assertEquals("Primary shards were not active [shards=1, active=0]", exception.getMessage());
+        try {
+            TimeValue timeout = TimeValue.timeValueMillis(between(10, 100));
+            UnavailableShardsException exception = expectThrows(
+                UnavailableShardsException.class,
+                () -> client().execute(
+                    GetGlobalCheckpointsAction.INSTANCE,
+                    new GetGlobalCheckpointsAction.Request("throttled-during-creation", true, true, EMPTY_ARRAY, timeout)
+                ).actionGet()
+            );
+            assertEquals(
+                "Primary shards were not active within timeout [timeout=" + timeout + ", shards=1, active=0]",
+                exception.getMessage()
+            );
+        } finally {
+            client().admin()
+                .cluster()
+                .prepareUpdateSettings()
+                .setPersistentSettings(
+                    Settings.builder().putNull(CLUSTER_ROUTING_ALLOCATION_NODE_INITIAL_PRIMARIES_RECOVERIES_SETTING.getKey()).build()
+                )
+                .get();
+        }
```

Comment:
Inline diff is totally confusing, consider reviewing in a split mode.

In short, this test:

* sets `cluster.routing.allocation.node_initial_primaries_recoveries=0` so that every created index is throttled
* creates an index
* executes `GetGlobalCheckpointsAction` with a timeout
* asserts it is complaining about unavailable shards rather then anything else
* reverts the setting to a default value

Reply:
I think I could also rewrite it to revert the setting after executing `GetGlobalCheckpointsAction` and assert it could get result back but I am not sure it would make the test better

Reply:
Yes I think we want that. As it stands the test suite still passes on `master` today (i.e. reverting the production-code changes in this PR). I think we have to check the success case here to see the value in this change.